sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/RedundantCast.cs

```diff
@@ -106,19 +106,24 @@ private static void CheckExtensionMethodInvocation(SyntaxNodeAnalysisContext con
                 return;
             }
 
-            var elementType = GetElementType(invocation, methodSymbol, context.SemanticModel);
-            if (elementType == null)
+            if (!(methodSymbol.ReturnType is INamedTypeSymbol returnType) ||
+                !returnType.TypeArguments.Any())
             {
                 return;
             }
 
-            if (!(methodSymbol.ReturnType is INamedTypeSymbol returnType) ||
-                !returnType.TypeArguments.Any())
+            var castType = returnType.TypeArguments.First();
+            if (methodSymbol.Name == "OfType"
+                && (castType.IsReferenceType || castType.Name == "Nullable"))
```

Comment:
Extracting a method with this condition (`IsOfTypeCallOnValueType`) will make the code easier to read.

Reply:
well, it's not on a value type, it's on a nullable type (either nullable value type or reference type).

Do you really think such a method would make the code more readable? `IsOfTypeOnNullableType`?

Reply:
Indeed we can extract just `castType.IsReferenceType || castType.Name == "Nullable"` as `IsNullableType`

Reply:
I chose `CanHaveNullValue` because `IsNullableType` is ambiguous (`castType.Name == "Nullable"` would suffice)